log4j-core/src/main/java/org/apache/logging/log4j/core/impl/ThrowableProxyHelper.java

```diff
@@ -85,7 +85,7 @@ static ExtendedStackTraceElement[] toExtendedStackTrace(
             stackLength = stackTrace.length;
         }
         final ExtendedStackTraceElement[] extStackTrace = new ExtendedStackTraceElement[stackLength];
-        Class<?> clazz = stack.isEmpty() ? null : stack.peek();
+        Class<?> clazz = stack.isEmpty() ? null : stack.peekLast();
```

Comment:
[`Deque#peekLast()`](https://docs.oracle.com/javase/8/docs/api/java/util/Deque.html#peekLast--) already returns `null` if the deque is empty, so the check is not necessary.

Reply:
<img width="1407" alt="Screen Shot 2023-01-17 at 2 43 45 PM" src="https://user-images.githubusercontent.com/36453467/213029221-bf8ab0c5-07f1-47d3-99c3-92c6326e635e.png">
<img width="1683" alt="Screen Shot 2023-01-17 at 2 46 02 PM" src="https://user-images.githubusercontent.com/36453467/213029231-6d9dc2cf-dfe7-4cca-940d-3b48c5817437.png">
Order of elements in stack did not change between 2.17.1 and 2.19.0 at least from the index number point of view.
I might have misunderstood you, sorry about that. Would like me to change the order of stack elements and preserve the current code in `ThrowableProxyHelper` or keep my changes to helper and don't change `StackLocator` as I assume these changed are mutually exclusive.
I might need some time to develop a unit test as there is no tests for helper class so far

Reply:
Yes, I was referring to the broken slow path, which still uses `push` instead of `addLast`.

Also:

* `stack.isEmpty() ? null : stack.peekLast()` can be replaced with `stack.peekLast()`,
* the comment in `org.apache.logging.log4j.util.PrivateSecurityManagerStackTraceUtil` states _"The size of the Deque is the number of methods on the execution stack. The first element is the class that started execution on this thread..."_ and is clearly not up-to-date.

Testing the slow path is not easy: you need to create a test named `*IT` that has a security manager that does not allow the "createSecurityManager" permission. See `EnvironmentPropertySourceSecurityManagerIT` for inspiration.